### PR TITLE
Added system_lib for system prestage jars

### PR DESCRIPTION
**Pull request by annaibm**
- Prestage system test prequisities to system_lib

related: https://github.com/adoptium/aqa-tests/issues/4912

### CODE REVIEW

**Comment by annaibm** [comment]
Grinder links: 
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/39879/consoleFull
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/39882/console
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/39906/consoleFull

**Comment by llxia** [comment]
pre-stage jars are not used. asm.jar still got downloaded to `/home/jenkins/workspace/Grinder/aqa-tests/systemtest_prereqs/asm/`:

```
00:08:53.766  configure-asm:
00:08:53.766       [echo] Executing macro download-file
00:08:53.766       [echo] File to download: https://repository.ow2.org/nexus/content/repositories/releases/org/ow2/asm/asm/9.0/asm-9.0.jar
00:08:53.766       [echo] Destination: /home/jenkins/workspace/Grinder/aqa-tests/systemtest_prereqs/asm/asm.jar
00:08:53.766       [echo] Download tool: curl
00:08:53.766      [mkdir] Created dir: /home/jenkins/workspace/Grinder/aqa-tests/systemtest_prereqs/asm
00:08:53.766       [exec]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
00:08:53.766       [exec]                                  Dload  Upload   Total   Spent    Left  Speed
00:08:53.766       [exec] 
00:08:54.639       [exec]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100  118k  100  118k    0     0   114k      0  0:00:01  0:00:01 --:--:--  114k
00:08:54.639  
```

I suspect that we need to update the reference in system test repo.

File: system/system.mk [hunk 1]
```diff
@@ -71,15 +71,21 @@ ifdef JVM_OPTIONS
   APPLICATION_OPTIONS := $(APPLICATION_OPTIONS) -jvmArgs $(Q)$(JVM_OPTIONS)$(Q)
 endif
 
+ifndef SYSTEM_LIB_DIR
+  SYSTEM_LIB_DIR := "/home/jenkins/workspace/Grinder/aqa-tests/systemtest_prereqs"
```
Comment (llxia) [inline system/system.mk:75]: Try to set `SYSTEM_LIB_DIR :=$(SYSTEMTEST_RESROOT)/systemtest_prereqs`

File: system/common.xml [hunk 2]
```diff
@@ -337,7 +343,7 @@
 				</then>
 		</if>
 		<copy todir="${SYSTEMTEST_DEST}/systemtest_prereqs/">
-			<fileset dir="${TEST_ROOT}/systemtest_prereqs/" includes="**" />
+			<fileset dir="${system_lib_dir}" includes="**"/>
```
Comment (llxia) [inline system/common.xml:346]: Why do we still need to copy to `systemtest_prereqs` dir?

File: buildenv/jenkins/JenkinsfileBase [hunk 3]
```diff
@@ -656,11 +663,18 @@ def buildTest() {
...
+				def customUrl = (env.BUILD_LIST == 'system') ? systemUrl : defaultUrl
 				if (PLATFORM.contains("windows")) {
 					env.LIB_DIR = env.LIB_DIR.replaceAll("\\\\", "/")
 				}
-				sh "perl ./aqa-tests/TKG/scripts/getDependencies.pl -path ${env.LIB_DIR} -task default -customUrl ${customUrl}"
+				if (env.BUILD_LIST != 'system') {
+					sh "perl ./aqa-tests/TKG/scripts/getDependencies.pl -path ${env.LIB_DIR} -task default -customUrl ${customUrl}"
+				} else {
+					sh "perl ./aqa-tests/TKG/scripts/getDependencies.pl -path ${env.LIB_DIR} -task default"
+				}
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:677]: for system tests, we do not need customUrl?

File: buildenv/jenkins/JenkinsfileBase [hunk 4]
```diff
@@ -656,11 +663,18 @@ def buildTest() {
 			// check for each lib. If lib does not exist, donwload it.
 			// If lib exists, SHA will be checked. Re-download if SHA does not match.
 			timeout(time: 20, unit: 'MINUTES') {
-				def customUrl = "https://ci.adoptium.net/job/test.getDependency/lastSuccessfulBuild/artifact/"
+				def defaultUrl = "https://ci.adoptium.net/job/test.getDependency/lastSuccessfulBuild/artifact/"
+				def systemUrl = "https://ci.adoptium.net/job/systemtest.getDependency/lastSuccessfulBuild/artifact/"
+				env.LIB_DIR = (env.BUILD_LIST == 'system') ? env.SYSTEM_LIB_DIR : env.LIB_DIR
+				def customUrl = (env.BUILD_LIST == 'system') ? systemUrl : defaultUrl
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:669]: I would prefer to have if statement to make the logic clear:

```
if (env.BUILD_LIST == 'system') {
   env.LIB_DIR = env.SYSTEM_LIB_DIR
   customUrl = "https://ci.adoptium.net/job/systemtest.getDependency/lastSuccessfulBuild/artifact/"
} else {
   customUrl = "https://ci.adoptium.net/job/test.getDependency/lastSuccessfulBuild/artifact/"
}

```

Please note this change will conflict with https://github.com/adoptium/aqa-tests/pull/5289. Depends on which PR gets merged first, the other one will need to be updated.

File: system/common.xml (hunk 2 above)
Comment (annaibm) [reply system/common.xml:346]: No, we don't need to copy to the `systemtest_prereqs` directory. I had kept it to check if local tests might need it; apparently, it's not needed. I will remove it.

File: buildenv/jenkins/JenkinsfileBase (hunk 3 above)
Comment (annaibm) [reply buildenv/jenkins/JenkinsfileBase:674]: Passing the customUrl for system tests was duplicating the process of downloading the jar files.

**Comment by annaibm** [comment]
Grinder links
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40707 (windows), 
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40708/

File: system/system.mk [hunk 5]
```diff
@@ -39,7 +39,9 @@ ifeq ($(CYGWIN),1)
   export PATH:=$(PERL):$(PATH)
 endif
 
+$(info TEST_RESROOT is $(TEST_RESROOT))
 SYSTEMTEST_RESROOT=$(TEST_RESROOT)/../
+$(info SYSTEMTEST_RESROOT is $(SYSTEMTEST_RESROOT))
```
Comment (llxia) [inline system/system.mk:44]: Please remove line 42 & Line 44. The value is set at a later stage.

File: system/system.mk [hunk 6]
```diff
@@ -71,15 +73,21 @@ ifdef JVM_OPTIONS
   APPLICATION_OPTIONS := $(APPLICATION_OPTIONS) -jvmArgs $(Q)$(JVM_OPTIONS)$(Q)
 endif
 
+ifndef SYSTEM_LIB_DIR
+  SYSTEM_LIB_DIR=$(SYSTEMTEST_RESROOT)/systemtest_prereqs
+endif
+
+$(info SYSTEM_LIB_DIR is $(SYSTEM_LIB_DIR))
```
Comment (llxia) [inline system/system.mk:80]: Please remove line 80. The value is set at a later stage.

File: buildenv/jenkins/JenkinsfileBase (hunk 3 above)
Comment (llxia) [reply buildenv/jenkins/JenkinsfileBase:676]: why having customUrl will duplicate the process? When running at openj9 Jenkins, we should try to use systemtest.getDependency from openj9 Jenkins, not Adoptium.

**Comment by llxia** [comment]
@annaibm can you point me to the code of creating `systemtest_prereqs` dir if it does not exist?

**Comment by annaibm** [comment]
> @annaibm can you point me to the code of creating `systemtest_prereqs` dir if it does not exist?

@llxia I have reverted the code to include copy of `systemtest_prereqs` as in  https://github.com/annaibm/aqa-tests/blob/test_systemtest/system/common.xml#L339C3-L341C10.

**Comment by llxia** [comment]
The current PR does not have `The systemtest-prereqs directory  could not be found. ` error locally?

**Comment by annaibm** [comment]
> The current PR does not have `The systemtest-prereqs directory could not be found. ` error locally?

No, it is getting processed as below
```
TESTING:
STF 08:58:26.042 - =========================   S T F   =========================
systemtest-prereqs has been processed, and set to: /home/jenkins/workspace/Grinder/jvmtest/system/systemtest_prereqsRetrieving amount of free space on drive containing /home/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_17163935052911/MiniMix_5m_0
There is 203270 Mb free
```

File: buildenv/jenkins/JenkinsfileBase (hunk 3 above)
Comment (annaibm) [reply buildenv/jenkins/JenkinsfileBase:673]: @llxia I have used customUrl and replaced `test.getDependency` to `systemtest.getDependency` for system test libs in TKG.

**Comment by annaibm** [comment]
Grinder links:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40841/

File: buildenv/jenkins/JenkinsfileBase (hunk 3 above)
Comment (annaibm) [reply buildenv/jenkins/JenkinsfileBase:673]: Grinder links:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40841/

**Comment by llxia** [comment]
https://github.com/adoptium/aqa-tests/blob/a6dbf7c16f535f74c0802d0340c5acf6333217ae/buildenv/jenkins/JenkinsfileBase#L679-L689 should be removed.

File: buildenv/jenkins/JenkinsfileBase [hunk 7]
```diff
@@ -656,29 +661,21 @@ def buildTest() {
 			// check for each lib. If lib does not exist, donwload it.
 			// If lib exists, SHA will be checked. Re-download if SHA does not match.
 			timeout(time: 20, unit: 'MINUTES') {
-				def customUrl = getCustomUrl()
-				if (PLATFORM.contains("windows")) {
-					env.LIB_DIR = env.LIB_DIR.replaceAll("\\\\", "/")
+					def customUrl = getCustomUrl()
+					if (PLATFORM.contains("windows")) {
```
Comment (LongyuZhang) [inline buildenv/jenkins/JenkinsfileBase:665]: The indention here needs to be adjusted.

File: system/common.xml [hunk 8]
```diff
@@ -336,9 +336,20 @@
 					<chmod file="${SYSTEMTEST_DEST}/openj9-systemtest/openj9.test.sharedClasses.jvmti/bin/native/**" perm="755"/>
 				</then>
 		</if>
-		<copy todir="${SYSTEMTEST_DEST}/systemtest_prereqs/">
-			<fileset dir="${TEST_ROOT}/systemtest_prereqs/" includes="**" />
-		</copy>
+		<condition property="system_lib_dir_not_set">
```
Comment (LongyuZhang) [inline system/common.xml:339]: Value should be set for `system_lib_dir_not_set`, either true or false

**Comment by annaibm** [comment]
Grinder links:
Sytem test
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40892/ 
non -system test: 
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40893/

**Comment by llxia** [comment]
This PR has to be merged with https://github.com/adoptium/TKG/pull/533 and https://github.com/adoptium/STF/pull/135

**Review by LongyuZhang — APPROVED** [review]
LGTM

**Pull request merged**